### DISCUSSION TITLE
How to use an N64 memory card mpk file on a PC emulator to test file validaty

### DISCUSSION THREAD

**Discussion by benmerrick3d** [sanni/cartreader#519] [Need help]
Hi all.

I have an mpk save from a memory card using the reader. I have checked the file online and it seems to show it is valid, but I want t try the file in a PC emulator to check its working.

It's a Perfect Dark save and I'm trying retro arch using default settings. I've placed the mpk file in the standard save folder and renamed it to the save as the game with mpk on the end but it doesn't show.

What am I doing wrong? I can share the file if it helps for tests. Normal save files work for me, just not sure what to do with memory card saves. 

Thank you

**Comment by WindyBison** [comment]
I do not know about retro arch, but for Project 64 on Windows when I have emulated games that require a Memory Pak, it saves it in the following folder:

`C:\Program Files (x86)\Project64 3.0\Save\`

I fired up Turok on the emulator and completed a save file

`TUROK_DINOSAUR_HUNTE_Cont_1.mpk`

When I hex dump this file, it looks good, but I don't have a Memory Pak for my physical N64 and Turok cartridge to test a save on a physical memory pak for comparison.

So given this I would try:

1. Read the ROM and Read the save file.
2. Load the rom in Project 64 and complete a Perfect Dark save.
3. Copy the newly created project64 mpk save file to a different directory just for comparison later if needed
4. Copy the physically read mpk file and overwrite the mpk file in the project64 save directory
5. Load the ROM again and see if your save is there

If Project64 is not reading the physically read mpk file, I suspect that there might be some modification to the save file needed much like the saves pulled off N64 Cartridges. I am not sure if there is a tool that exists for the memory pak saves, but the wiki for this github does have a tool that can do EEPROM, SRAM, and FLASH save conversion. I don't know if it also supports the memory pak.

**Reply by benmerrick3d** [reply]
Thanks for the reply, I appreciate the help. I will have a look
